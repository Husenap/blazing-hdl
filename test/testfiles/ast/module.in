interface MyFloat {
  sign : bool
  exp : uint
  mant : uint
}

mod ModuleWithFloat (
  x -> MyFloat
  y <- MyFloat
) { }

interface RvChannel {
  data <- uint
  valid <- bool
  ready -> bool
}

interface Something {
  AR <- RvChannel
  AW <- RvChannel
  R -> RvChannel
  W <- RvChannel
  B -> RvChannel
}


