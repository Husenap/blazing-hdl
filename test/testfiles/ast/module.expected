InterfaceStmt: MyFloat
  Ports: 
    PortStmt: undirected bool sign
    PortStmt: undirected uint exp
    PortStmt: undirected uint mant
ModStmt: ModuleWithFloat
  Ports: 
    PortStmt: output MyFloat x
    PortStmt: input MyFloat y
  Body: 
InterfaceStmt: RvChannel
  Ports: 
    PortStmt: input uint data
    PortStmt: input bool valid
    PortStmt: output bool ready
InterfaceStmt: Something
  Ports: 
    PortStmt: input RvChannel AR
    PortStmt: input RvChannel AW
    PortStmt: output RvChannel R
    PortStmt: input RvChannel W
    PortStmt: output RvChannel B
